doc/read-the-docs-site/adr/0006-indexer-resuming-strategy.rst

```diff
@@ -45,16 +45,88 @@ Decision
   In the current Marconi interface, it is actually just ``securityParam + 1`` as we assume that all rollbackable blocks *can* be fully stored on disk because of the disk flush.
   We assume that all the data contained in the indexer is consistent for all the point returned by an indexer.
 
-* We will have each indexer start the node-to-client chain-sync protocol at different points in time.
-  However, the ``Coordinator`` will make sure that all indexers are synced until a common point, and advance at the same speed (i.e. they can only request the next block once all the indexers have finished processing the current block).
+* We will start a single node chain-sync client for all indexers instead of one node chain sync client per indexer.
 
-* We will use the full set of points provided by ``resumableFromStorage`` of an indexer as resuming points for the node-to-client chain-sync protocol.
-  These points need to ordered in descending order so that protocol can priotise the selection of the latest ones.
+* We will implement the following resuming logic.
+
+  In order to start the chain-sync client, we need to provide to provide intersection points (or resuming points).
+  Intersection points represent the points from which we want to start syncing from the local node.
+  We will provide as intersection points the points returned from the following pseudo-code:
+
+  .. code-block:: haskell
+
+    -- | Query the resumable points of each indexer then select the resumable
+    -- points of the indexer that has the lowest immutable point. If two indexer
+    -- have the same lowest immutable point, select the one which has the lowest
+    -- overall point (immutable or not).
+    selectIntersectionPoints indexers = do
+      resumingPoints <- mapM queryResumingPoints indexers
+      head $ sortOn (\rps -> (maximum $ filter isImmutablePoint rps, maximum rps)) resumingPoints
```

Comment:
Is this correct? It doesn't look like an intersection. Actually do we want to return an intersection at all? ie. if one indexer is behind, the intersection is empty but we want to restart from there. I don't know what would be the correct set of points to send... all the points from the least advanced indexer?

Reply:
Sorry, the name is wrong. It's intersection with the points from the local node :) 

So it's as you said, the pseudo-code says that we return all the resuming points from the indexer that has the lowest immutable resume point.

Reply:
Don't we need to intersect it with other indexers? Hmm... maybe not, we'll see when we'll implement it. Though, it should be `minimum`, not `maximum`, right?